src/engine/Map.cxx

```diff
@@ -331,9 +331,7 @@ std::vector<uint8_t> Map::calculateAutotileBitmask(const Point &isoCoordinates)
              mapNodes[it.first * m_columns + it.second]->getMapNodeDataForLayer(currentLayer).tileData &&
              mapNodes[x * m_columns + y] && mapNodes[x * m_columns + y]->getMapNodeDataForLayer(currentLayer).tileData &&
              ((mapNodes[it.first * m_columns + it.second]->getMapNodeDataForLayer(currentLayer).tileID ==
-               mapNodes[x * m_columns + y]->getMapNodeDataForLayer(currentLayer).tileID) ||
-              (mapNodes[it.first * m_columns + it.second]->isLayerAutoTile(currentLayer) &&
-               isLayerAutoTile(isoCoordinates, currentLayer)))))
+               mapNodes[x * m_columns + y]->getMapNodeDataForLayer(currentLayer).tileID))))
```

Comment:
your should know the history here:

the removed check **originally** checke if both tiles are from the TileType `AutoTile` . @Mograbi changed this in his last PR, and added a function "IsLayerAutoTile" because we now have more layers that autotile. (Roads, Autotoile, Underground, ...) 

This check is not correct, because it autotiles everything together that can autotile, but only tiles of the same TileID should do that.
Except roads. All roads should autotile to each other.

The function checks for
```cpp
    if (tileData->tileType == +TileType::ROAD || tileData->tileType == +TileType::AUTOTILE ||
        tileData->tileType == +TileType::UNDERGROUND)
```